action.yml

```diff
@@ -29,27 +32,60 @@ outputs:
 runs:
   using: composite
   steps:
+    - id: registryconfig
+      if: ${{ cloudbees.registries != '' }}
+      name: Resolve source image
+      uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/services/registry-config:0.0.22
+      with:
+        entrypoint: /registry-config
+        args: --config="${{ inputs.registry-config }}" resolve "${{ inputs.src }}" resolved-image.file
+
     - id: cranecopy
       name: Copy a remote image from a given source to a destination
       uses: docker://gcr.io/go-containerregistry/crane:debug
       run: |
-        set -euo pipefail
+        set -uo pipefail
 
-        if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
-          echo "Validating source image ${{ inputs.src }}"
-          crane validate --remote ${{ inputs.src }} --platform ${{ inputs.platform }}
+        if [ ! -e resolved-image.file ]; then
+          echo '${{ inputs.src }}' > resolved-image.file
         fi
 
-        echo "Copying source image ${{ inputs.src }} to destination ${{ inputs.destination }}"
-        crane copy ${{ inputs.src }} ${{ inputs.destination }} --platform ${{ inputs.platform }}
+        set +e
+
+        while IFS= read -r line; do
+          if [ "$SKIP_IMAGE_VALIDATION" == false ]; then
+            echo "Validating source image ${line}"
+            crane validate --remote ${line} --platform ${{ inputs.platform }}
+
+            if [ $? -ne 0 ]; then
+              echo "Failed to validate image: $line"
+              continue
+            fi
+          fi
+
+          echo "Copying source image ${line} to destination ${{ inputs.destination }}"
```

Comment:
The `crane copy` command does not need to be in the loop, if a mirror is not available, the `crane validate` command would be able to verify that, right? Therefore we should be able to move this command outside the loop.

Reply:
nvm the `validate` command is optional.

Reply:
Still can be beneficial if copy fails for some reason. So it can have another try if multiple registries.